### PR TITLE
Fresher sphinx gallery

### PR DESCRIPTION
**Pull request by corinnebosley**
I have hijacked this from @choldgraf so that I can continue with the PR without pestering him to do too much more work on it, as he has done a lot of work already and we haven't been keeping up, so it's time we did.

@pp-mo This should just need a rebase (maybe?) and a few tweaks.  Hoping to get this merged today or tomorrow.

### CODE REVIEW

**Comment by corinnebosley** [comment]
WHAT?! Stickler is fast becoming me nemesis.  It had no trouble finding the stickler.yml file an hour ago, it's not disappeared since then, so why can't you find it now?!!

File: docs/source/conf.py [hunk 1]
```diff
@@ -45,19 +45,20 @@
...
-              'cartopy.sphinxext.gallery',
               'sphinx.ext.autodoc',
               'sphinx.ext.doctest',
               'sphinx.ext.intersphinx',
               'sphinx.ext.coverage',
               'sphinx.ext.viewcode',
               'sphinx.ext.extlinks',
-              #'sphinxcontrib.napoleon', (Needs work before this can be enabled.)
+              # 'sphinx.ext.autosummary',
+              # 'sphinxcontrib.napoleon', (Needs work before this can be enabled.)
```
[the remark this answers is not in the document]
Comment (corinnebosley) [reply docs/source/conf.py]: Oh now you can find the yml file can you?  How convenient, and unsettling...

**Comment by corinnebosley** [comment]
@choldgraf Ready to go with this if you're happy?  

Will get @pp-mo to merge ASAP (which will probably be tomorrow morning now).

**Comment by choldgraf** [comment]
hey @corinnebosley , thanks for hijacking, I appreciate the time you're putting in to pushing this over the finish line :-)  

Can you give a general idea of what you've changed from the original PR? I don't see my commits in there but maybe you have just copy/pasted? Let me know!

**Comment by choldgraf** [comment]
(also sorry for slowness of response but I'm on california time, so just getting started for the day!)

**Comment by corinnebosley** [comment]
Hi @choldgraf, I figured you must have been working to a different time zone, no worries. I got pretty confused with branching off your branch and rebasing, but I think I ended up fixing up most of your commits except the first one.  Other than that, I just made a few minor changes; mostly pep8 but I also commented out the autosummary option and tweaked an import which had been broken somehow.

**Comment by corinnebosley** [comment]
@pp-mo Can we get this merged today please?

**Comment by choldgraf** [comment]
I'm +1 on a merge. Minor tweaks / problems can always be spot-checked. Thanks for the work pushing this through @corinnebosley :-)

**Comment by pp-mo** [comment]
This has developed a merge problem from a change on master.
So I made a rebased version + made ready to PR+merge it, 

but now I've stumbled on a question ...

Within the 'extensions' settings in `conf.py`, this PR adds a reference to `sphinx_gallery.gen_gallery`, and thus accordingly (I think) **removes** the reference to `cartopy.sphinxext.gallery` and hence the call to `cartopy.sphinxext.gallery.gen_gallery`

However ... 
 * (1) doesn't that mean that `cartopy.sphinxext.gallery` is now entirely obsolete + should be removed ?, ***or alternatively . . .***
 * (2) doesn't it mean there is now ***also*** no invocation of  `cartopy.sphinxext.gallery.gen_examples()` either, and isn't that a mistake ?

**Comment by corinnebosley** [comment]
@pp-mo I've removed the file ```cartopy.sphinxext.gallery``` and pulled out some obsolete references and that tag.  Tested the build, which seemed to work fine.

One thing that I'm a little confused about is that I removed refs to 'gallery' and 'examples/*' from the toctree in ```cartopy.docs.source.index.rst```, and I'm not sure if it needs to be replaced with something...

**Comment by corinnebosley** [comment]
Thanks @choldgraf, I think that's working now, although I'm still not entirely sure, because it also looks a lot like the reference itself has just disappeared... I may come back to this.

**Comment by corinnebosley** [comment]
In the meantime, this is still not quite ready for merge, there are still a couple of HTTP errors I'm getting at build time that look like this:

```
Embedding documentation hyperlinks in examples ...
WARNING: The following HTTP Error has occurred: 404
WARNING: The following HTTP Error has occurred: 403
```
So I would like to find these and fix them, and maybe also rearrange the gallery structure so that the URLs don't have to be changed and will still link up with Stack Overflow and whatnot.

**Comment by corinnebosley** [comment]
@pp-mo Have at it.

**Comment by pp-mo** [comment]
> rearrange the gallery structure so that the URLs don't have to be changed 

On review, looking at [previous comment](https://github.com/SciTools/cartopy/pull/902#discussion_r144064618), I don't think that is possible because the 'old' structure just isn't the way that sphinx-gallery works -- it just *requires* subfolders*, if you want example categories ([probably relevant link?](https://sphinx-gallery.github.io/getting_started.html#the-structure-of-the-examples-folder)).

**Comment by pp-mo** [comment]
When building the docs, we now get a whole bunch of new warnings of this sort :
```
/home/h05/itpp/git/cartopy/cartopy_main/lib/cartopy/io/img_tiles.py:docstring of cartopy.io.img_tiles.GoogleTiles:13: WARNING: Unexpected indentation.
/home/h05/itpp/git/cartopy/cartopy_main/lib/cartopy/io/img_tiles.py:docstring of cartopy.io.img_tiles.GoogleTiles.tile_bbox:5: WARNING: Unexpected section title.
```
Those all seem to be fairly benign formatting errors.
Also, fixing them here would over-complicate this PR, 
so **I think these can be fixed later**.
( I am also considering putting a docs build into the CI testing, so that ought to show up outstanding problems of this nature )

We also have a couple of new warning messages in another part of the build :
```
embedding documentation hyperlinks...
WARNING: The following HTTP Error has occurred fetching https://docs.scipy.org/doc/numpy/reference/searchindex.js: 404 Not Found
WARNING: The following HTTP Error has occurred fetching http://toblerity.org/shapely/: 403 Forbidden
```
By hacking sphinx-gallery for extra debug, I find that these are failures to link into the documentation of related projects sphinx-docs.
I.E. these are triggered in relation to config entries in `docs/source/conf.py'.
This means that links to numpy and shapely in the docs aren't working.
However, comparing to the older docs, it seems these weren't working before anyway,
**so I think this can also be ignored**

**Comment by pp-mo** [comment]
Thanks @choldgraf, @corinnebosley + respect for sticking with it !

**Pull request merged**